File: GMS-backend/GMS/Entities/Endpoint/Trener/Edit/TreneriEditEndpoint.cs
using GMS.Data;
using GMS.Helpers;
using GMS.Helpers.Auth;
using Microsoft.AspNetCore.Mvc;

namespace GMS.Entities.Endpoint.Trener.Edit
{
    [Route("Trener-Edit")]
    [MyAuthorization]

    public class TreneriEditEndpoint : MyBaseEndpoint<TreneriEditRequest, int>
    {
        private readonly ApplicationDbContext db;

        public TreneriEditEndpoint(ApplicationDbContext db)
        {
            this.db = db;
        }

        [HttpPost]
        public override async Task<int> Handle([FromBody]TreneriEditRequest request, CancellationToken cancellationToken)
        {
            Models.Trener? trener;
            if (request.ID == 0)
            {
                trener = new Models.Trener();
                db.Add(trener);


            }
            else
            {
                trener = db.Trener.FirstOrDefault(s => s.ID == request.ID);
                if (trener == null)
                    throw new Exception("pogresan ID");
            }

            trener.Ime = request.Ime.RemoveTags();
            trener.Prezime = request.Prezime.RemoveTags();
            trener.BrojTelefona = request.BrojTelefona.RemoveTags();
            trener.Slika = request.Slika?.RemoveTags();
 

            await db.SaveChangesAsync(cancellationToken);

            return trener.ID;
        }
    }
}

File: GMS-frontend/src/app/endpoints/treneri-endpoints/treneri-edit-endpoint.ts

import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {Config} from "../../config";
import {Observable} from "rxjs";
@Injectable({providedIn: 'root'})
export class TreneriEditEndpoint implements  MyBaseEndpoint<TreneriEditRequest, number>{
  constructor(public httpClient:HttpClient) { }

  Handle(request: TreneriEditRequest): Observable<number> {
    let url=Config.adresa + `Trener-Edit`;

    return this.httpClient.post<number>(url, request);
  }

}
export interface TreneriEditRequest {
  id: number
  ime: string
  prezime: string
  brojTelefona: string
  slika: string | undefined
}

